src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,8 +1,70 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+    private static final String RESULT = "result";
+    private int supply;
+    private int buy;
+    private int result;
+    private StringBuilder stringBuilder = new StringBuilder();
+    private String report;
 
     public void getStatistic(String fromFileName, String toFileName) {
+        readInfo(fromFileName);
+        calculateResult();
+        createReport();
+        writeInfoToReport(report, toFileName);
+    }
+
+    private void readInfo(String fromFile) {
+        try {
+            BufferedReader reader = new BufferedReader(new FileReader(fromFile));
+            String value = reader.readLine();
+            while (value != null) {
+                stringBuilder.append(value).append(System.lineSeparator());
+                value = reader.readLine();
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read file:" + fromFile, e);
+        }
+    }
+
+    private void calculateResult() {
+        buy = 0;
+        supply = 0;
+        result = 0;
+        String[] info = stringBuilder.toString().split(System.lineSeparator());
+        for (String str : info) {
+            if (str.substring(0, str.indexOf(",")).equals(SUPPLY)) {
+                supply += Integer.parseInt(str.substring(str.indexOf(",") + 1));
+            } else if (str.substring(0, str.indexOf(",")).equals(BUY)) {
+                buy += Integer.parseInt(str.substring(str.indexOf(",") + 1));
```

Comment:
instead of these substrings, lets split `line` into an array and access data by index